xdsl/transforms/individual_rewrite.py

```diff
@@ -0,0 +1,37 @@
+from dataclasses import dataclass
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import PatternRewriter, RewritePattern
+from xdsl.tools.command_line_tool import get_all_dialects
+from xdsl.traits import HasCanonicalisationPatternsTrait
+
+REWRITE_BY_NAMES: dict[str, dict[str, RewritePattern]] = {
+    op.name: {
+        pattern.__class__.__name__: pattern
+        for pattern in trait.get_canonicalization_patterns()
+    }
+    for dialect in get_all_dialects().values()
+    for op in dialect().operations
+    if (trait := op.get_trait(HasCanonicalisationPatternsTrait)) is not None
+}
+
+
+@dataclass
+class IndividualRewrite(ModulePass):
+    name = "individual-rewrite"
```

Comment:
I'm not a big fan of the name, usually we have verbs to express what a pass is doing?
Like `apply-individual-rewrite`?